hazelcast-client-legacy/src/test/java/com/hazelcast/client/cache/nearcache/ClientCacheSerializationCountTest.java

```diff
@@ -0,0 +1,266 @@
+            public Portable create(int classId) {
+                return new SerializationCountingData();
+            }
+        });
+    }
+
+    @Test
+    public void testDeserializationCountWith_ObjectNearCache_cacheLocalUpdatePolicy() {
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(InMemoryFormat.OBJECT
+                , NearCacheConfig.LocalUpdatePolicy.CACHE);
+        prepareCache(nearCacheConfig);
+
+        final String key = randomString();
+        SerializationCountingData value = new SerializationCountingData();
+        cache.put(key, value);
+        assertAndReset(1, 0);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                assertEquals(1, nearCache.size());
+            }
+        });
+
+        cache.get(key);
+        assertAndReset(0, 0);
+
+        cache.get(key);
+        assertAndReset(0, 0);
+    }
+
+    @Test
+    public void testDeserializationCountWith_BinaryNearCache_cacheLocalUpdatePolicy() {
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(InMemoryFormat.BINARY
+                , NearCacheConfig.LocalUpdatePolicy.CACHE);
+        prepareCache(nearCacheConfig);
+
+        String key = randomString();
+        SerializationCountingData value = new SerializationCountingData();
+        cache.put(key, value);
+        assertAndReset(1, 0);
+
+        cache.get(key);
+        assertAndReset(0, 1);
+
+        cache.get(key);
+        assertAndReset(0, 1);
+    }
+
+    @Test
+    public void testDeserializationCountWith_ObjectNearCache_invalidateLocalUpdatePolicy() {
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(InMemoryFormat.OBJECT
+                , NearCacheConfig.LocalUpdatePolicy.INVALIDATE);
+        prepareCache(nearCacheConfig);
+
+        String key = randomString();
+        SerializationCountingData value = new SerializationCountingData();
+        cache.put(key, value);
+        assertAndReset(1, 0);
+
+        cache.get(key);
+        assertAndReset(0, 1);
+
+        cache.get(key);
+        assertAndReset(0, 0);
+    }
+
+    @Test
+    public void testDeserializationCountWith_BinaryNearCache_invalidateLocalUpdatePolicy() {
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(InMemoryFormat.BINARY
+                , NearCacheConfig.LocalUpdatePolicy.INVALIDATE);
+        prepareCache(nearCacheConfig);
+
+        String key = randomString();
+        SerializationCountingData value = new SerializationCountingData();
+        cache.put(key, value);
+        assertAndReset(1, 0);
+
+        cache.get(key);
+        assertAndReset(0, 1);
+
+        cache.get(key);
+        assertAndReset(0, 1);
+    }
+
+    @Test
+    public void testDeserializationCountWithoutNearCache() {
+        prepareCache(null);
+
+        SerializationCountingData value = new SerializationCountingData();
+
+        String key = randomString();
+        cache.put(key, value);
```

Comment:
I think, we need assertEquals(1, nearCache.size()); check here also because we put to near-cache on put to cache